cdap-metadata-ext-spanner/src/main/java/io/cdap/cdap/metadata/spanner/TableUpdates.java

```diff
@@ -0,0 +1,293 @@
+  private final String namespace;
+  private final String type;
+  private final String name;
+  private final Long created;
+  private final StringBuilder userText;
+  private final StringBuilder systemText;
+  private final Set<Property> metadataProps;
+
+  /**
+   * Constructs and processes all metadata for a given entity.
+   *
+   * @param entity the metadata entity being processed.
+   * @param metadata the metadata to be associated with the entity.
+   * @throws IOException if schema parsing fails.
+   */
+  public TableUpdates(MetadataEntity entity, Metadata metadata) throws IOException {
+    this.entity = entity;
+    this.namespace = entity.getValue("namespace");
+    this.type = entity.getType().toLowerCase();
+    this.name = Objects.requireNonNull(entity.getValue(entity.getType())).toLowerCase();
+
+    Map<ScopedName, String> processedProperties = new HashMap<>();
+    Set<Property> processedPropsWithSummary = processAndSummarizeProperties(metadata.getProperties(),
+                                                                            processedProperties);
+    Set<Property> extractedTags = extractTags(metadata.getTags());
+    this.metadataProps = new HashSet<>();
+    this.metadataProps.add(new Property(MetadataScope.SYSTEM.name(), this.type, this.name));
+    this.metadataProps.addAll(processedPropsWithSummary);
+    this.metadataProps.addAll(extractedTags);
+    this.userText = new StringBuilder(buildText(processedProperties, metadata.getTags(), MetadataScope.USER));
+    this.systemText = new StringBuilder(Stream.of(type, buildText(processedProperties, metadata.getTags(),
+                                                                  MetadataScope.SYSTEM)).filter(s -> !s.isEmpty())
+                                          .collect(Collectors.joining(" ")).trim());
+    this.created = validateCreationTime(processedProperties).orElse(null);
+  }
+
+  public String getNamespace() {
+    return namespace;
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public Optional<Long> getCreated() {
+    return Optional.ofNullable(created);
+  }
+
+  public String getUserText() {
+    return userText.toString().trim();
+  }
+
+  public String getSystemText() {
+    return systemText.toString().trim();
+  }
+
+  public Set<Property> getMetadataProps() {
+    return metadataProps;
+  }
+
+  /**
+   * Processes raw properties, creates a summary property, and extracts them into a Set.
+   * This method has a side effect: it populates the processedProperties map.
+   *
+   * @param rawProperties The original properties from the metadata object.
+   * @param processedProperties Output parameter: A map that will be populated with processed values.
+   * @return A Set containing processed Property objects and a single summary property.
+   */
+  private Set<Property> processAndSummarizeProperties(Map<ScopedName, String> rawProperties,
```

Comment:
Now you can just pass `processedProperties` and avoid calling `reformatSchemaProperty` here.